cyberdrop_dl/scraper/crawlers/coomer_crawler.py

```diff
@@ -250,26 +227,108 @@ async def get_user_info(self, scrape_item: ScrapeItem) -> dict:
+            api_url (URL): The legacy API URL.
+            new_properties (dict): Properties from the current response.
+            cached_properties (dict): Properties from the cached response.
+            response: The latest HTTP response to save to the cache.
+
+        Returns:
+            int: The updated maximum offset.
+        """
+        user_str = new_properties.get("name", "Unknown")
+        new_count = int(new_properties.get("count", 0))
+        cached_count = int(cached_properties.get("count", 0))
+        if cached_count == 0:
+            return
+        post_limit = int(new_properties.get("limit", 50))
+        shift = new_count - cached_count
+
+        if shift > 0:
+            log(f"{shift} new posts detected for {user_str} (Coomer). Adjusting cache...", 20)
+
+            cached_posts = []
+            offset = 0
+            while True:
+                paginated_api_url = api_url.with_query({"o": offset})
+                cache_key = self.manager.cache_manager.request_cache.create_key("GET", paginated_api_url)
+                cached_response = await self.manager.cache_manager.request_cache.get_response(cache_key)
+
+                if not cached_response:
+                    break  # Stop if no cached response exists for the current offset
+
+                cached_json = await cached_response.json()
+                cached_posts.extend(cached_json)
+                offset += post_limit
+
+            all_posts = ["placeholder"] * shift + cached_posts
+            new_pages = [all_posts[i : i + post_limit] for i in range(0, len(all_posts), post_limit)]
+
+            for page_index, page_posts in enumerate(new_pages):
+                offset = page_index * post_limit
+                paginated_api_url = api_url.with_query({"o": offset})
+                cache_key = self.manager.cache_manager.request_cache.create_key("GET", paginated_api_url)
+
+                if "placeholder" in page_posts:
+                    # Invalidate cache for this page
+                    await self.manager.cache_manager.request_cache.delete_url(cache_key)
+                    log(f"Invalidated cached page: {paginated_api_url}", 20)
+                else:
+                    cached_response = await self.manager.cache_manager.request_cache.get_response(cache_key)
+                    if cached_response:
+                        # Update the cached response
+                        adjusted_content = json.dumps(page_posts).encode("utf-8")
+                        cached_response._body = adjusted_content
+                        cached_response.reset()
+
+                        await self.manager.cache_manager.request_cache.save_response(
+                            cached_response,
+                            cache_key,
+                            datetime.datetime.now()
+                            + self.manager.config_manager.global_settings_data.rate_limiting_options.file_host_cache_length,
+                        )
+
```

Comment:
After shifting, it's possible for the last cached response to have less than `post_limit` number of posts, even though it's not the last page. This may happen if the user stopped CDL in the middle of a scrape.

If `len(page_posts)` is less than `post_limit` but `offset` != `max_offset`, invalidate `cached_response` as well

Reply:
Good catch, will implement!